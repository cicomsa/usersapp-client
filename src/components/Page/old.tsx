import React from 'react'
import { RouteComponentProps } from 'react-router-dom'
import Users from '../Users'
import { IDataContent, IData } from '../../interfaces'
import { TParams, TUser } from '../../types'
const page = require('../data.json')

const Page = ({ match }: RouteComponentProps<TParams>) => {
  const { id }: TParams = match.params
  const { data }: IData = page

  const content: IDataContent | undefined = data.find(item => item.id === Number(id))

  if (!content) {
    return <p>Error retreiving data for this page</p>
  }

  let title: string = content ? content.title : ''
  let description: string = content ? content.description : ''
  let users: TUser[] | any[] = content ? content.users : []

  return (
    <>
      <h2>{title}</h2>
      <p>Description: {description}</p>
      <Users users={users} />
    </>
  )
}