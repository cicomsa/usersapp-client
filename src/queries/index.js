import gql from 'graphql-tag'

const GET_PAGE = gql`
  query page($param: ID!) {
    page(id: $param) {
      id
      title,
      description,
      users {
        id,
        name,
        email,
        location,
        online,
        avatar
      }
    }
  }
`

export {
  GET_PAGE
}
