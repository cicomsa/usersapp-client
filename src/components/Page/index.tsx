import React from 'react'
import { RouteComponentProps } from 'react-router-dom'
import { useQuery } from '@apollo/react-hooks'
import Users from '../Users'
import { GET_PAGE } from '../../queries'
import { TParams, TPageQueryResponse, TPageQueryVariables } from '../../types'

const Page = ({ match }: RouteComponentProps<TParams>) => {
  const { id }: TParams = match.params
  const param: string = (Number(id) - 1).toString()

  const { loading, data, error } = useQuery<TPageQueryResponse, TPageQueryVariables>(
    GET_PAGE,
    {
      variables: { param }
    }
  )

  if (loading) return <p>Loading...</p>;
  if (error) return <p>Error retrieving data from the server.</p>;

  if (data) {
    return (
      <>
        <h2 className="page-title">{data.page.title}</h2>
        <p>Description: {data.page.description}</p>
        <Users users={data.page.users} />
      </>
    )
  }

  return null
}

export default Page
