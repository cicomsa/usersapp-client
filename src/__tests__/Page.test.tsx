import React from 'react'
import { ApolloProvider } from 'react-apollo';
import { create, act } from 'react-test-renderer'
import Page from '../components/Page'
import { match } from 'react-router'
import wait from 'waait'
import { GET_PAGE } from '../queries'
import { MockedProvider } from '@apollo/react-testing';
const data = require('../components/data.json')

const path: string = '/route/:id'

const matchParam: match<{ id: string }> = {
  isExact: false,
  path,
  url: path.replace(':id', '1'),
  params: { id: "1" }
};

const routeComponentPropsMock = {
  history: {} as any,
  location: {} as any,
  match: { params: { id: '1' } } as any
}

const routeComponentPropsMockError = {
  history: {} as any,
  location: {} as any,
  match: { params: { id: '5' } } as any
}

const mocks = [
  {
    request: {
      query: GET_PAGE,
      variables: { param: '0' },
    },
    result: {
      data: {
        page: data.page[0]
      },
    }
  }
]

const mocksError = [
  {
    request: {
      query: GET_PAGE,
      variables: { param: '4' },
    },
    error: new Error('error')
  }
]

describe('Page', () => {
  it('renders loading state initially', () => {
    let wrapper: any
    act(() => {
      wrapper = create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Page {...routeComponentPropsMock} />
        </MockedProvider>
      )
    })

    expect(wrapper.toJSON()).toMatchSnapshot()

    const el = wrapper.root.findByType('p')
    expect(el.children).toContain('Loading...')
  })
  it('renders Page content', async () => {
    let wrapper: any
    act(() => {
      wrapper = create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Page {...routeComponentPropsMock} />
        </MockedProvider>
      )
    })

    expect(wrapper.toJSON()).toMatchSnapshot();

    await act(async () => {
      await wait(0)

      await wrapper.update(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Page {...routeComponentPropsMock} />
        </MockedProvider>
      )
    })

    expect(wrapper.toJSON()).toMatchSnapshot()

    const el = wrapper.root.findByType('h2')
    expect(wrapper.root.findByType('h2').children).toContain('Page One')
  })

  it('renders error message', async () => {
    let wrapper: any
    act(() => {
      wrapper = create(
        <MockedProvider mocks={mocks} addTypename={false}>
          <Page {...routeComponentPropsMock} />
        </MockedProvider>
      )
    })

    expect(wrapper.toJSON()).toMatchSnapshot();

    await act(async () => {
      await wrapper.update(
        <MockedProvider mocks={mocksError} addTypename={false}>
          <Page {...routeComponentPropsMockError} />
        </MockedProvider>
      )
    })

    expect(wrapper.toJSON()).toMatchSnapshot()

    const el = wrapper.root.findByType('p')
    expect(el.props.children).toContain('Error')
  })
})
