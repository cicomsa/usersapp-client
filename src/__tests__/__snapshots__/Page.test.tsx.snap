// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders Page content 1`] = `
<p>
  Loading...
</p>
`;

exports[`Page renders Page content 2`] = `
Array [
  <h2
    className="page-title"
  >
    Page One
  </h2>,
  <p>
    Description: 
    List of users from UK
  </p>,
  <table
    className="users-table"
  >
    <thead>
      <tr>
        <th
          className="table-header"
        >
          Name
        </th>
        <th
          className="table-header"
        >
          Email
        </th>
        <th
          className="table-header"
        >
          Location
        </th>
        <th
          className="table-header"
        >
          Online
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="user-content "
        >
          <img
            alt="avatar"
            className="user-img"
            src="https://source.unsplash.com/random/10x10"
          />
          John Moe
        </td>
        <td
          className="user-content "
        >
          <a
            href="mailto:john_moe@email.com"
          >
            john_moe@email.com
          </a>
        </td>
        <td
          className="user-content "
        >
          London
        </td>
        <td
          className="user-content user-online"
        />
      </tr>
      <tr>
        <td
          className="user-content "
        >
          <img
            alt="avatar"
            className="user-img"
            src="https://source.unsplash.com/random/10x10"
          />
          Jane Doe
        </td>
        <td
          className="user-content "
        >
          <a
            href="mailto:jane_doe@email.com"
          >
            jane_doe@email.com
          </a>
        </td>
        <td
          className="user-content "
        >
          London
        </td>
        <td
          className="user-content user-online"
        />
      </tr>
      <tr>
        <td
          className="user-content "
        >
          <img
            alt="avatar"
            className="user-img"
            src="https://source.unsplash.com/random/10x10"
          />
          Mike Smith
        </td>
        <td
          className="user-content "
        >
          <a
            href="mailto:mike_smith@email.com"
          >
            mike_smith@email.com
          </a>
        </td>
        <td
          className="user-content "
        >
          Brighton
        </td>
        <td
          className="user-content user-online"
        >
          <svg
            className="svg-icon"
            viewBox="0 0 20 20"
          >
            <path
              d="M10.219,1.688c-4.471,0-8.094,3.623-8.094,8.094s3.623,8.094,8.094,8.094s8.094-3.623,8.094-8.094S14.689,1.688,10.219,1.688 M10.219,17.022c-3.994,0-7.242-3.247-7.242-7.241c0-3.994,3.248-7.242,7.242-7.242c3.994,0,7.241,3.248,7.241,7.242C17.46,13.775,14.213,17.022,10.219,17.022 M15.099,7.03c-0.167-0.167-0.438-0.167-0.604,0.002L9.062,12.48l-2.269-2.277c-0.166-0.167-0.437-0.167-0.603,0c-0.166,0.166-0.168,0.437-0.002,0.603l2.573,2.578c0.079,0.08,0.188,0.125,0.3,0.125s0.222-0.045,0.303-0.125l5.736-5.751C15.268,7.466,15.265,7.196,15.099,7.03"
            />
          </svg>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;

exports[`Page renders error message 1`] = `
<p>
  Loading...
</p>
`;

exports[`Page renders error message 2`] = `
<p>
  Error retrieving data from the server.
</p>
`;

exports[`Page renders loading state initially 1`] = `
<p>
  Loading...
</p>
`;
